libs/state/file/store/src/lib/providers/upload-file.service.ts

```
@@ -0,0 +1,113 @@
+import { StoryBlock } from './../../../../../../model/convs-mgr/stories/blocks/main/src/lib/story-block.interface';
+import { Injectable } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { AngularFireStorage } from '@angular/fire/compat/storage';
+import { AngularFirestore } from '@angular/fire/compat/firestore';
+
+import { Repository, DataService } from '@ngfi/angular';
+
+import { of } from 'rxjs';
+import { finalize, switchMap, map, tap, take } from 'rxjs/operators';
+
+import { Logger } from '@iote/bricks-angular';
+
+import { StoryBlocksStore } from '@app/state/convs-mgr/stories/blocks';
+
+import { Organisation } from '@app/model/organisation'
+import { StoryBlockTypes } from '@app/model/convs-mgr/stories/blocks/main';
+import { FileMessageBlock } from '@app/model/convs-mgr/stories/blocks/messaging';
+
+import { ActiveOrgStore } from '@app/state/organisation';
+
+import { FileUpload } from '../model/file-upload.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UploadFileService {
+  protected store = 'file-store';
+
+  protected _activeRepo: Repository<FileUpload>;
+  org: Organisation;
+  imageLink: string;
+
+
+
+  constructor(
+    private _storyBlockService$: StoryBlocksStore,
+    private _org$$: ActiveOrgStore,
+    private _repoFac: DataService,
+    _logger: Logger,
+    private _ngfiStorage: AngularFireStorage,
+    private document: AngularFirestore) { }
+
+
+
+  upload(block: FileUpload, url: string, blockType: StoryBlockTypes, form: FormGroup, id: string) {
+
+
+    const fileBlock:FileUpload = {
+      fileId: id,
+      filePath: url,
+      fileType: blockType,
+      size: '3MB'
+    };
+
+    this._org$$.get().pipe(
+      map((org) => {
+        this.org = org;
+        return (!!org ?? false)
```

Throw error if org is null instead of letting the flow continue